fe/fe-core/src/main/java/com/starrocks/analysis/CreateFunctionStmt.java

```diff
@@ -195,11 +254,72 @@ private void analyzeUdf() throws AnalysisException {
         String closeFnSymbol = properties.get(CLOSE_SYMBOL_KEY);
         function = ScalarFunction.createUdf(
                 functionName, argsDef.getArgTypes(),
-                returnType.getType(), argsDef.isVariadic(),
+                returnType.getType(), argsDef.isVariadic(), TFunctionBinaryType.HIVE,
                 objectFile, symbol, prepareFnSymbol, closeFnSymbol);
         function.setChecksum(checksum);
     }
 
+    private void checkStarrocksJarUdfType(Type type, Class ptype, String pname) throws AnalysisException {
+        if (!(type instanceof ScalarType)) {
+            throw new AnalysisException("UDF does not support non-scalar type: " + type);
+        }
+        ScalarType scalarType = (ScalarType) type;
+        Class cls = PrimitiveTypeToJavaClassType.get(scalarType.getPrimitiveType());
+        if (cls == null) {
+            throw new AnalysisException("UDF does not support type: " + scalarType);
+        }
+        if (!cls.equals(ptype)) {
+            throw new AnalysisException(String.format("UDF %s[%s] type does not match %s", pname,
+                    ptype.getCanonicalName(), cls.getCanonicalName()));
+        }
+    }
+
+    private void checkStarrocksJarUdfMethod(Method method) throws AnalysisException {
+        String name = method.getName();
+        if (PREPARE_METHOD_NAME.equals(name) || EVAL_METHOD_NAME.equals(name)) {
+            Class retType = method.getReturnType();
+            checkStarrocksJarUdfType(returnType.getType(), retType, "Return");
+            if (method.getParameters().length != argsDef.getArgTypes().length) {
+                throw new AnalysisException(String.format("UDF '%s' parameter count does not match", name));
+            }
+            for (int i = 0; i < method.getParameters().length; i++) {
+                Parameter p = method.getParameters()[i];
+                checkStarrocksJarUdfType(argsDef.getArgTypes()[i], p.getType(), p.getName());
+            }
+        } else if (CLOSE_METHOD_NAME.equals(name)) {
+            Class retType = method.getReturnType();
+            if (!retType.equals(void.class)) {
+                throw new AnalysisException(String.format("UDF '%s' return type should be void", CLOSE_METHOD_NAME));
+            }
+            if (method.getParameters().length != 0) {
+                throw new AnalysisException(String.format("UDF '%s' should have zero parameter", CLOSE_METHOD_NAME));
+            }
+        }
+    }
+
+    private void checkStarrocksJarUdfClass() throws AnalysisException {
+        boolean hasEvalMethod = false;
+        for (Method m : udfClass.getMethods()) {
+            if (EVAL_METHOD_NAME.equals(m.getName())) {
```

Comment:
should we consider limiting the number of evaluate functions in the UDF to 1

Reply:
But if more than 1 evaluate function, then function signature won't match.

Reply:
I aggree with you, check evaluate function in one class is unnecessary

Reply:
But check one `evaluate` method will make it easier for @stdpain to write BE code.

Reply:
I guess it's because for BE code, it only needs to check method name, instead of construcing full function signature.

Reply:
I don't JNI call method not need params type... It's really? @stdpain

Reply:
According to my knowldege, Java supports function overload. So there will be two eveluate method:
1. evaluate(int)
2.  evaluate(float)
we have to pick right signature.

Reply:
Yeah, you right, and if you are JNI develeper, you will choose method only accorder by method name?

Reply:
Limiting the number of evaluate methods will make it easier for BE to obtain function signatures